sale_order_mass_action/wizards/sale_order_mass_action_wizard.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2015-Today GRAP (http://www.grap.coop)
+# @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrderMassActionWizard(models.TransientModel):
+
+    _name = "sale.order.mass.action.wizard"
+
+    confirm = fields.Boolean(
+        help="Check this box if you want to confirm all the selected quotations."
+    )
+
+    def _get_sale_order_confirm_domain(self):
+        return [
+            ("id", "in", self.env.context.get("active_ids")),
+            ("state", "in", ("draft", "sent")),
+        ]
+
+    def apply_button(self):
+        sale_order_obj = self.env["sale.order"]
+        if self.env.context.get("active_model") != "sale.order":
+            return
+        for wizard in self.filtered("confirm"):
+            sale_orders = sale_order_obj.search(wizard._get_sale_order_confirm_domain())
+            sale_orders.action_confirm()
```

Comment:
as a improvment we could add something like : 
```if len(sale_orders) != len(context.get("active_ids"):
       self.env.user.notify_warning(message='The following orders has not been confirmed, because there was in an incorrect status %s' % ...)
```

(depends on OCA/web/web_notify).

Reply:
Indeed, I can do it, it's better for UX

Reply:
@legalsylvain I'm not sure on this as I need to search more sale orders to display them.

But I can at least notify for the success